Packages/com.unity.inputsystem/InputSystem/Plugins/InputForUI/InputSystemProvider.cs

```diff
@@ -62,30 +66,45 @@ private static void Bootstrap()
         private EventModifiers _eventModifiers => _inputEventPartialProvider._eventModifiers;
 
         private DiscreteTime _currentTime => (DiscreteTime)Time.timeAsRational;
-        
+
         private const uint kDefaultPlayerId = 0u;
 
         public void Initialize()
         {
             _inputEventPartialProvider ??= new InputEventPartialProvider();
             _inputEventPartialProvider.Initialize();
-            
+
             _events.Clear();
-            
+
             _mouseState.Reset();
             _seenMouseEvents = false;
 
             _penState.Reset();
             _seenPenEvents = false;
-            
+
             _touchState.Reset();
             _seenTouchEvents = false;
-            
+
             // TODO should UITK somehow override this?
             _cfg = Configuration.GetDefaultConfiguration();
             RegisterActions(_cfg);
         }
 
+        private void OnNextPreviousPerformed(InputAction.CallbackContext ctx)
+        {
+            if (ctx.control.device is Keyboard)
+            {
+                var keyboard = ctx.control.device as Keyboard;
+                EventProvider.Dispatch(Event.From(new NavigationEvent
+                {
+                    type = NavigationEvent.Type.Move,
+                    direction = keyboard.shiftKey.isPressed ? NavigationEvent.Direction.Previous : NavigationEvent.Direction.Next,
```

Comment:
Hm, smart!

Reply:
Should these have repeat rate as well?
Also don't use EventProvider.Dispatch here, I've added DispatchFromCallback so we can sort events later as order of callbacks is not really defined

Reply:
For Next and Previous too it would make sense, since they're normally tied to KeyDown events which also repeat.

Reply:
Thanks @benoit-alain , will implement it in a follow up PR.